### PR TITLE
Harden types

### LINKED ISSUE #43: Try to harden the types even more

### PR DESCRIPTION
**Pull request by DanielBadura**
fixes #43

### CODE REVIEW

File: src/Store/MultiTableStore.php [hunk 1]
```diff
@@ -39,7 +40,7 @@ public function __construct(
     /**
      * @param class-string<AggregateRoot> $aggregate
      *
-     * @return AggregateChanged[]
+     * @return array<AggregateChanged>
```
Comment (DavidBadura) [inline src/Store/MultiTableStore.php:43]:
```suggestion
     * @return list<AggregateChanged>
```

File: src/Store/DoctrineStore.php [hunk 2]
```diff
@@ -28,27 +29,37 @@ public function connection(): Connection
...
         if (!$recordedOn instanceof DateTimeImmutable) {
             throw new StoreException('recordedOn should be a DateTimeImmutable object');
         }
 
-        $result['recordedOn'] = $recordedOn;
+        return $recordedOn;
+    }
 
-        return $result;
+    private static function normalizePlayhead(string $playheadAsString, AbstractPlatform $platform): int
```
Comment (DavidBadura) [inline src/Store/DoctrineStore.php:55]: simple int cast should be fine here

File: src/Store/MultiTableStore.php [hunk 3]
```diff
@@ -96,7 +97,7 @@ public function has(string $aggregate, string $id): bool
 
     /**
      * @param class-string<AggregateRoot> $aggregate
-     * @param AggregateChanged[]          $events
+     * @param array<AggregateChanged>     $events
```
Comment (DavidBadura) [inline src/Store/MultiTableStore.php:100]:
```suggestion
     * @param list<AggregateChanged> $events
```

File: src/Store/MultiTableStore.php (hunk 1 above)
Comment (DanielBadura) [reply src/Store/MultiTableStore.php:43]: cant be done unless using `array_values` after `array_merge`. Seems not feasible for this

File: src/Store/MultiTableStore.php (hunk 3 above)
Comment (DanielBadura) [reply src/Store/MultiTableStore.php:100]: cant be done unless using `array_values` after `array_merge`. Seems not feasible for this

**Pull request merged**